http-clients/java-nio-client/src/test/java/software/amazon/awssdk/http/nio/java/internal/S3JavaHttpClientIntegratedTest.java

```diff
@@ -138,22 +171,19 @@ private static void getObjectTest(S3AsyncClient s3Async) {
     }
 
     private static void deleteObjectTest(S3AsyncClient s3Async) {
-        CompletableFuture<DeleteObjectResponse> future = s3Async.deleteObject(DeleteObjectRequest.builder()
-                .bucket(BUCKET)
-                .key(KEY)
-                .build());
-        future.whenComplete((resp, err) -> {
-            try {
-                if (resp != null) {
-                    System.out.println("my response: " + resp);
-                } else {
-                    // Handle error
-                    err.printStackTrace();
-                }
-            } finally {
-                // Lets the application shut down. Only close the client when you are completely done with it.
-                s3Async.close();
-            }
-        }).join();
+        try (s3Async) {
+            CompletableFuture<DeleteObjectResponse> future = s3Async.deleteObject(
+                    DeleteObjectRequest.builder()
+                            .bucket(BUCKET)
+                            .key(KEY)
+                            .build());
+            future.whenComplete((resp, err) -> {
```

Comment:
Just remove this complete. `join()` will throw if there's an error and the test will fail which is what we want.